pkg/descheduler/controllers/migration/controller.go

```diff
@@ -0,0 +1,701 @@
+	if err != nil {
+		return
+	}
+	for i := range jobList.Items {
+		v := &jobList.Items[i]
+		timeoutDuration := 30 * time.Minute
+		if v.Spec.TTL != nil && v.Spec.TTL.Duration > 0 {
+			timeoutDuration = v.Spec.TTL.Duration + 5*time.Minute
+		}
+		if r.clock.Since(v.CreationTimestamp.Time) < timeoutDuration {
+			continue
+		}
+		if err := r.deleteReservation(context.TODO(), v); err != nil {
+			if !errors.IsNotFound(err) {
+				break
+			}
+		}
+		err = r.Client.Delete(context.TODO(), v)
+		if err != nil {
+			klog.Errorf("Failed to scavenge PodMigrationJob %s, err: %v", v.Name, err)
+		} else {
+			klog.V(4).Infof("Successfully scavenge PodMigrationJob %s", v.Name)
+		}
+	}
+}
+
+// +kubebuilder:rbac:groups=scheduling.koordinator.sh,resources=podmigrationjobs,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=scheduling.koordinator.sh,resources=podmigrationjobs/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=scheduling.koordinator.sh,resources=reservations,verbs=get;list;watch;create;update;patch;delete
+
+// Reconcile reads that state of the cluster for a PodMigrationJob object and makes changes based on the state read
+// and what is in the Spec
+// Automatically generate RBAC rules to allow the Controller to read and write Deployments
+func (r *Reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	job := &sev1alpha1.PodMigrationJob{}
+	err := r.Client.Get(ctx, request.NamespacedName, job)
+	if errors.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	}
+	if err != nil {
+		klog.Errorf("Failed to Get PodMigrationJob from %v, err: %v", request, err)
+		return reconcile.Result{}, err
+	}
+
+	if !r.assumedCache.isNewOrSameObj(job) {
+		return reconcile.Result{}, nil
+	}
+
+	result, err := r.doMigrate(ctx, job)
+	if err != nil {
+		klog.Errorf("Failed to reconcile MigrationJob %v, err: %v", request.NamespacedName, err)
+	}
+	r.assumedCache.assume(job)
+	return result, err
+}
+
+func (r *Reconciler) doMigrate(ctx context.Context, job *sev1alpha1.PodMigrationJob) (reconcile.Result, error) {
+	klog.V(4).Infof("begin process MigrationJob %s", job.Name)
+	if job.Spec.Paused {
+		return reconcile.Result{}, nil
+	}
+
+	if job.Status.Phase != "" &&
+		job.Status.Phase != sev1alpha1.PodMigrationJobPending &&
+		job.Status.Phase != sev1alpha1.PodMigrationJobRunning {
+		return reconcile.Result{}, nil
+	}
+
+	timeout, err := r.abortJobIfTimeout(ctx, job)
+	if err != nil {
+		return reconcile.Result{}, err
+	} else if timeout {
+		return reconcile.Result{}, nil
+	}
+
+	if job.Status.Phase == "" || job.Status.Phase == sev1alpha1.PodMigrationJobPending {
+		job.Status.Phase = sev1alpha1.PodMigrationJobRunning
+	}
+
+	if job.Spec.Mode == sev1alpha1.PodMigrationJobModeEvictionDirectly {
+		return r.evictPodDirectly(ctx, job)
+	}
+
+	if job.Spec.ReservationOptions == nil || job.Spec.ReservationOptions.ReservationRef == nil {
+		err = r.createReservation(ctx, job)
```

Comment:
create reservation succeed, the next reconcile will be triggered by reservation event?

Reply:
Currently, MigrationController used polling mechanism instead of checking the status of the Reservation and updating the PodMigrationJob in linkage with the events associated with the Reservation.  Controller checked status every 3 seconds by default. The implementation of this part can be optimized in the future.